app/helpers/repository_helper.rb

```diff
@@ -1,16 +1,162 @@
+  # object will not be commited and so should be commited later on by the caller.
+  #
+  # Returns a tuple containing a boolean and an array. If the +txn+ argument was nil, the boolean
+  # indicates whether the transaction was completed without errors. If the +txn+ argument was not
+  # nil, the boolean indicates whether any errors were encountered which mean the caller should not
+  # commit the transaction. The array contains any error or warning messages as arrays of arguments
+  # (each of which can be passed directly to flash_message from a controller).
+  def remove_files(file_revisions, user, repo, path: '/', txn: nil)
+    messages = []
+
+    if txn.nil?
+      txn = repo.get_transaction(user.user_name)
+      commit_txn = true
+    else
+      commit_txn = false
+    end
+
+    current_path = Pathname.new path
+
+    current_revision = repo.get_latest_revision.revision_identifier
+
+    file_revisions.each do |file_path, revision|
+      subdir_path, basename = File.split(file_path)
+      basename = sanitize_file_name(basename)
+      file_path = current_path.join(subdir_path).join(basename)
+      file_path = file_path.to_s
+      txn.remove(file_path, revision || current_revision)
+    end
+
+    if commit_txn
+      success, txn_messages = commit_transaction repo, txn
+      [success, messages + txn_messages]
+    else
+      [true, messages]
+    end
+  end
+
+  # Helper method that commits a transaction +txn+ in repo +repo+. Returns a boolean and an array
+  # where the boolean indicates whether the transaction was commited successfully and an array
+  # containing any error or warning messages.
+  #
+  # Does not attempt to commit the transaction if there are no jobs present to commit.
+  def commit_transaction(repo, txn)
+    return [false, [:no_files]] unless txn.has_jobs?
+    return [true, []] if repo.commit(txn)
+
+    [false, [:txn_conflicts, txn.conflicts]]
+  end
+
+  def flash_repository_messages(messages, suppress: nil)
+    suppress ||= {}
+    messages.each do |msg, other_info|
+      next if suppress[msg]
+      case msg
+      when :too_large
+        max_size = (MarkusConfigurator.markus_config_max_file_size / 1_000_000.00).round(2)
+        flash_message(:error, I18n.t('student.submission.file_too_large',
+                                     file_name: other_info,
+                                     max_size: max_size))
+      when :too_small
+        flash_message(:warning, I18n.t('student.submission.empty_file_warning', file_name: other_info))
+      when :invalid_filename
+        flash_message(:error, I18n.t('student.submission.invalid_file_name'))
+      when :extra_files
+        flash_message(:error, I18n.t('assignments.upload_file_requirement'))
+      when :no_files
+        flash_message(:warning, I18n.t('student.submission.no_action_detected'))
+      when :txn_conflicts
+        flash_message(:error, partial: 'submissions/file_conflicts_list', locals: { conflicts: other_info })
+      else
```

Comment:
This is interesting; was it a style thing? If not, I'd prefer remove the `else... next` entirely.

Reply:
Yeah this was to avoid another style warning.  However I agree that it's redundant in this [case](https://www.wordnik.com/words/no%20pun%20intended). I'll remove it.